concepts/rxjs-primer.md

```diff
@@ -0,0 +1,521 @@
+promise based API, but that is really just scratching the surface:
+
+```js
+fromEvent(button, 'click')
+  .pipe(
+    mergeMap(value => {
+      // this 'inner' subscription is managed by mergeMap, with response value emitted to observer
+      return makeHttpRequest(value);
+    })
+  )
+  .subscribe(response => {
+    // do something
+  });
+```
+
+**We can then divide the flattening operators into common behaviors like...**
+
+#### Operators that `switch`
+
+Like a light switch, `switch` based operators will turn off (unsubscribe) the
+current observable and turn on a new observable on emissions from the source.
+Switch operators are useful in situations you don't want (or need) more than one
+active observable at a time:
+
+```js
+inputValueChanges
+  // only the last value is important, if new value comes through cancel previous request / observable
+  .pipe(
+    // make GET request for data
+    switchMap(requestObservable)
+  )
+  .subscribe();
+```
+
+Switch based operators include `switchAll`,
+[`switchMap`](../operators/transformation/switchmap.md), and `switchMapTo`.
+
+#### Operators that `concat`
+
+Comparable to a line at the ATM machine, the next transaction can't begin until
+the previous completes. In observable terms, only one subscription will occur at
+a time, in order, triggered by the completion of the previous. These are useful
+in situations where order of execution is important:
+
+```js
+concat(
+  firstObservable,
+  // will begin when 'firstObservable` completes
+  secondObservable,
+  // will begin when 'secondObservable` completes
+  thirdObservable
+).subscribe(values => {
+  // take action
+});
+```
+
+Merge based operators include [`concat`](../operators/combination/concat.md),
+[`concatAll`](../operators/combination/concatall.md),
+[`concatMap`](../operators/transformation/concatmap.md), and
+[`concatMapTo`](../operators/transformation/concatmapto.md).
+
+#### Operators that `merge`
+
+Like your merging lane on the interstate, merge based operators support multiple
+active observables flowing into one lane in a first come first serve basis.
+Merge operators are useful in situations where you want to trigger an action
+when an event from one of many sources occurs:
+
+```js
+merge(firstObservable, secondObservable)
+  // any emissions from first or second observable as they occur
+  .pipe(mergeMap(saveActivity))
+  .subscribe();
+```
+
+Merge based operators include [`merge`](../operators/combination/merge.md),
+[`mergeMap`](../operators/transformation/mergemap.md), `mergeMapTo` and
+[`mergeAll`](../operators/combination/mergeall.md).
+
+## Other similarities between operators
+
+There are also operators that share a similar goal but offer flexibility in
+their triggers. For instance, for unsubscribing from an observable after a
+specific condition is met, we could use:
+
+1. [`take`](../operators/filtering/take.md) when we know we only ever want `x`
+   values.
+2. `takeLast` when you just want the last value.
```

Comment:
Missing a link to the docs for this one?

Reply:
Realized I don’t have this one on the site yet 😳